pkg/internal/traces/read_decorator.go

```diff
@@ -84,8 +83,11 @@ func hostNamePIDDecorator(cfg *InstanceIDConfig) decorator {
 	}
 
 	// caching instance ID composition for speed and saving memory generation
-	return func(s *svc.Attrs, hostPID int) {
-		s.UID.Instance = fullHostName + ":" + strconv.Itoa(hostPID)
+	return func(s *svc.Attrs, _ int) {
+		// FIXME - the pid here is causing a cardinality explosion of the
+		// 'target_info' metrics when instrumenting processes that fork()
+		// s.UID.Instance = fullHostName + ":" + strconv.Itoa(hostPID)
+		s.UID.Instance = fullHostName
```

Comment:
This would make that multiple processes in the same host have the same instance ID, so it can screw prometheus handling of instances.